mlrun/datastore/sources.py

```diff
@@ -905,41 +927,32 @@ def __init__(
         )
 
     def to_dataframe(self):
```

Comment:
it is a bit misleading that the method is called to_dataframe but an `MongoDBIterator` can be returned instead of a `pandas.DataFrame`... try to give a better name and add type hints for the returning value:

```
def to_dataframe(self) -> Union[pd.DataFrame, _MondoDBIterator]:
```

Reply:
I cannot change the name because it part of the api.. @guy1992l, but I add the typing.